purchase_ext_sst/models/purchase_order.py

```diff
@@ -103,27 +103,29 @@ def open_record(self):
 
     @api.multi
     def write(self, vals):
-        val_phone_search = 'phone_search' in vals and vals['phone_search'] \
-                           or False
-        val_phone = 'phone_update' in vals and vals['phone_update'] or False
-        val_mobile = 'mobile_update' in vals and vals['mobile_update'] \
-                     or False
-        for order in self:
-            val_tent_name = 'tentative_name' in vals and vals[
-                'tentative_name'] or order.tentative_name
-            partner_id = 'partner_id' in vals and vals['partner_id']\
-                                 or order.partner_id.id
-            if self.is_default_partner(partner_id) and \
-                    (val_phone or val_mobile or val_phone_search):
-                partner_id = self.create_partner(val_phone or
-                                                 val_phone_search,
-                                                 val_mobile).id
-            if not self.is_default_partner(partner_id):
-                partner = self.env['res.partner'].browse(partner_id)
-                self.update_partner(partner, val_phone, val_mobile,
-                                    val_tent_name)
-        vals['partner_id'] = partner_id
-        vals['phone_update'] = vals['mobile_update'] = False
+        if 'phone_search' in vals or 'phone_update' in vals or \
```

Comment:
@yostashiro Do you think we should add `'partner_id' in vals` in here as well? I cannot imagine if there is cases that need this statement.

Reply:
@TimLai125 I'm a bit lost.  What issue are you trying to solve here?

Reply:
@yostashiro Here is used to solve the purchase order cancel issue. When multiple orders is selected in the tree view and use the "Cancel Purchase Orders" button, the `partner_id` of all `purchase.order` will be updated.